src/java.net.http/share/classes/jdk/internal/net/http/common/SSLFlowDelegate.java

```diff
@@ -73,17 +83,21 @@
  * --------->  data flow direction
  *
  *
- *                         +------------------+
- *        upstreamWriter   |                  | downWriter
- *        ---------------> |                  | ------------>
- *  obtained from this     |                  | supplied to constructor
- *                         | SSLFlowDelegate  |
- *        downReader       |                  | upstreamReader
- *        <--------------- |                  | <--------------
- * supplied to constructor |                  | obtained from this
- *                         +------------------+
+ *              |                                       |
+ *              |   upstreamWriter                      |   upstreamReader
+ *              |    obtained from                      |    obtained from
+ *              |   upstreamWriter()                    |   upstreamReader()
+ *              v                                       v
+ *      +-------------------+                   +-------------------+
+ *      *                   *                   *                   *
+ *      *  SSLFlowDelegate  *                   *  SSLFlowDelegate  *
+ *      *     encrypts      *                   *     decrypts      *
+ *      +-------------------+                   +-------------------+
+ *              |                                       |
+ *              |   downWriter                          |   downReader
+ *              | supplied to constructor               |  supplied to constructor
+ *              v                                       v
```

Comment:
Can we keep the original diagram too? The new diagram is very nice but gives the impression that encrypt/decrypt are completely independent threads of execution and they're not - since consuming some data may probe the engine to request sending of some data.

Reply:
Hello Daniel, I've updated the PR to slightly improve the current proposed form to address the part that the encrypt/decrypt are independent. To me, the old diagram wasn't clear enough on what represented upstream/downstream (given that the directions were horizontal), plus what kind of data the upstream/downstream received.

Do you think in the updated version of this PR, the diagram addresses this review comment? If not, I'll update the PR further to include the original diagram too.

Reply:
I think you will need to invert the right hand side of the new diagram: for reading, upstream is below, while for writing, upstream is above.

Reply:
I see what you mean. I've updated the PR accordingly.